### ISSUE TITLE
bug: not skipping 

### ISSUE THREAD

**Issue by rafapereirabr** [ipeaGIT/gtfs2gps#88]
Apparently, the `gtfs2gps` function is not skipping shape_ids with missing routes/trips. Here is a reproducible example of the problem when it hits the  shape_id `8700-21-0`

```
library(gtfs2gps)
library(magrittr)
library(data.table)
library(sf)

# local GTFS.zip
spo_zip <- system.file("extdata/saopaulo.zip", package="gtfs2gps" )

# read gtfs
spo_gtfs <- gtfs2gps::read_gtfs(spo_zip)

# subset time interval
spo_gtfs_f <- gtfs2gps::filter_day_period(spo_gtfs, period_start = "07:00:", period_end = "07:30")
  
# Convert GTFS data into a data.table with GPS-like records
spo_gps <- gtfs2gps2(spo_gtfs_f, spatial_resolution = 15, progress = T, cores = 1 )

```
> The following Trip_ids have been ignored due to missing data in original gtfs.zip: 8700-21-0
 Error in FUN(X[[i]], ...) : object 'departure_time' not found

**Comment by pedro-andrade-inpe** [comment]
The problem occurs in `filter_day_period`, as it works properly for the whole saopaulo data. There are two bugs:

1) In `filter_day_period()`, line 31, it is 
```R
  gtfs$stops <- gtfs$stops[ stop_id %in% stop_id ]
```

and should be

```R
  gtfs$stops <- gtfs$stops[ stop_id %in% unique_stops ]
```
isn´t it?

2) Filtering `stop_times` by time can create trips that have only one `stop_time`, which is inconsistent as a trip must have at least two `stop_times`. The implemented algorithm for `gtfs2gps` supposes each `trip` has at least two `stop_times`. Should such trips be removed by `filter_day_period`? Do you know an easy way to do that using `data.table`?

**Comment by rafapereirabr** [comment]
You're right. I've just fixed issue (1).

regarding issue (2), we should be able to discart trips with a single valid `departure_time` just as we already deal with 'invalid' trips with no valid `departure_time`.

**Comment by rafapereirabr** [comment]
note. the problem is in `shape_id` "52896", which has only one `trip_id` "8700-21-0"

**Comment by rafapereirabr** [comment]
I think I've fixed issue (2) with commit [34662d9](https://github.com/ipeaGIT/gtfs2gps/commit/34662d90be0ca5136490434fe67a74467f1d82d6).

The  the number of stops (`nstop`) is smaller than 2, the gtfs2gps function should ignore that shape_id. There are probably better ways to go around this problem and report the shape_ids that are being ignored, but this should do the work for now.

```
    # Get the stops sequence with lat long linked to that route
    # each shape_id only has one stop sequence
    nstop <- gtfs_data$stop_times[trip_id %in% all_tripids, .N, by ="trip_id"]$N
    
    # If there is less than two valid stops/departure times, jump this shape_id
    if ( nstop < 2) {   return(NULL) }
```

**Comment by rafapereirabr** [comment]
Now fully fixed with

```
    # If there is less than two valid stops, jump this shape_id
    if( min(nstop) < 2){ return(NULL) } # nocov

```
